config/modules/validator.py

```
@@ -0,0 +1,364 @@
+            if isinstance(value, str):
+                # Strip whitespace and validate non-empty
+                value = value.strip()
+                if not value:
+                    return False
+                try:
+                    # Handle both integer and float formats
+                    if value_type is int:
+                        # First convert to float to handle scientific notation
+                        value = int(float(value))
+                    else:
+                        value = float(value)
```

Ruff

Use ternary operator `value = int(float(value)) if value_type is int else float(value)` instead of `if`-`else`-block

(SIM108)